pykernels/graph/shortestpath.py

```diff
@@ -18,9 +18,9 @@ def floyd_warshall(adj_mat, weights):
     res = res + ((adj_mat != 0) * weights)
     res[res == 0] = np.inf
     np.fill_diagonal(res, 0)
-    for i in xrange(N):
-        for j in xrange(N):
-            for k in xrange(N):
+    for i in range(N):
```

Comment:
This will slow down a lot code for Python2. I would use range from six module (http://pythonhosted.org/six/) which defaults to xrange for Python2